MoreLinq.Test/AggregateRightTest.cs

```diff
@@ -0,0 +1,129 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using LinqEnumerable = System.Linq.Enumerable;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class AggregateRightTest
+    {
+        // Overload 1 Test
+
+        [Test]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void AggregateRight1WithNullSequence()
+        {
+            (null as IEnumerable<int>).AggregateRight((a, b) => a + b);
+        }
+
+        [Test]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void AggregateRight1WithNullFunc()
+        {
+            Enumerable.Range(1, 5).AggregateRight(null);
+        }
+
+        [Test]
+        [ExpectedException(typeof(InvalidOperationException))]
+        public void AggregateRight1WithEmptySequence()
+        {
+            new int[] { }.AggregateRight((a, b) => a + b);
+        }
+
+        [Test]
+        public void AggregateRight1SimpleTest()
+        {
+            var numbersAsString = Enumerable.Range(1, 5).Select(x => x.ToString());
+
+            Assert.IsTrue("(1+(2+(3+(4+5))))" == numbersAsString.AggregateRight((a, b) => string.Format("({0}+{1})", a, b)));
```

Comment:
Replace with `Assert.That(numbersAsString.AggregateRight((a, b) => string.Format("({0}+{1})", a, b)), Is.EqualTo("(1+(2+(3+(4+5))))")`.